src/workerd/api/kv.c++

```diff
@@ -147,8 +147,9 @@ jsg::Promise<KvNamespace::GetWithMetadataResult> KvNamespace::getWithMetadata(
   auto headers = kj::HttpHeaders(context.getHeaderTable());
   auto client = getHttpClient(context, headers, LimitEnforcer::KvOpType::GET, urlStr);
 
+  auto request = client->request(kj::HttpMethod::GET, urlStr, headers);
   return context.awaitIo(js,
-      client->request(kj::HttpMethod::GET, urlStr, headers).response,
+      kj::mv(request.response),
```

Comment:
This is probably because `client` is `kj::mv`ed as part of the same statement (see next line of code). It is unspecified in what order these happen, and compilers can differ. It seems when compiling for Windows, Clang is doing the `kj::mv` before it attempts to dereference `client`, and so by the time it dereferences, it is moved away.

I'm surprised that Clang's behavior differs between Linux and Windows, though. I would normally expect a particular compiler to be consistent across platforms. Maybe they felt they needed to match MSVC on Windows? I wonder if there's a flag we can use to control this. It's going to be pretty annoying for those of us developing on Linux to have to deal with these problems, which won't be noticed until CI time...